types/src/error.rs

```diff
@@ -16,12 +16,37 @@ impl<T: fmt::Display> fmt::Display for Mismatch<T> {
 	}
 }
 
+/// Invalid params.
+#[derive(Debug)]
+pub struct InvalidParams;
+
+/// Error that may occur when a server fails to execute provided closure/callback.
+#[derive(Debug, thiserror::Error)]
+pub enum ServerCallError {
+	#[error("Invalid params in the RPC call")]
+	/// InvalidParams,
+	InvalidParams(InvalidParams),
+	#[error("Provided context failed: {0}")]
+	/// Provided context/metadata failed.
+	ContextFailed(#[source] Box<dyn std::error::Error + Send + Sync>),
+}
+
+impl From<InvalidParams> for ServerCallError {
+	fn from(params: InvalidParams) -> Self {
+		Self::InvalidParams(params)
+	}
+}
+
 /// Error type.
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
+	/// Error that may occur when a server fails to execute provided closure/callback.
```

Comment:
The wording here seems to imply that these are errors that can happen when a closure/callback was provided, implicitly suggesting that there could be other ways to use the library. I think it'd be less easy to misunderstand if we just had "Error that may occur when a server fails to execute a call.".